### PR TITLE
Overload map() to support key paths

### PR DESCRIPTION
**Pull request by sharplet**
We should be able to use key paths with `flatMap()` too, but this is a start.

- [x] Updated CHANGELOG.md.

### CODE REVIEW

File: Tests/ReactiveSwiftTests/PropertySpec.swift [hunk 1]
```diff
@@ -732,6 +732,17 @@ class PropertySpec: QuickSpec {
 					property.value = 2
 					expect(mappedProperty.value) == 3
 				}
+
+#if swift(>=3.2)
+				it("should work with key paths") {
+					let property = MutableProperty("foo")
+					let mappedProperty = property.map(\.count)
```
Comment (NachoSoto) [inline Tests/ReactiveSwiftTests/PropertySpec.swift]: LOVE IT

**Comment by mdiep** [comment]
KeyPaths are the best thing ever. 💖

**Comment by andersio** [comment]
(Irrelevant to this PR) I hope the Swift gurus manage to get key path decomposition into 4.0. With it we can make `MutableProperty` capable of automatic KVO.

**Comment by mdiep** [comment]
I don't think these tests are actually going to run on CI right now. But since we've updated Nimble, we should be able to run `swift test` with Swift 4 on CI.

**Comment by sharplet** [comment]
Happy to wait for #436 and rebase.

**Comment by sharplet** [comment]
or #437 😂

**Comment by sharplet** [comment]
Ready to go 👍

**Comment by mdiep** [comment]
🎉

**Pull request merged**